src/history/hs_rec.c

```diff
@@ -452,6 +452,8 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_MULTI *mult
                 continue;
 
             if (F_ISSET(upd, WT_UPDATE_TO_DELETE_FROM_HS)) {
+                WT_ASSERT_ALWAYS(session, F_ISSET(upd, WT_UPDATE_HS | WT_UPDATE_RESTORED_FROM_HS),
+                  "the update must still be in the history store for us to delete.");
```

Comment:
Thanks for adding the assertion. This message could potentially come out in a MongoDB log file, so it would be useful to add more context. Perhaps:
"Attempting to remove an update from the history store in WiredTiger, but the update was missing"